Readability.js

```diff
@@ -844,18 +844,50 @@ Readability.prototype = {
 
         this._initializeNode(topCandidate);
       } else if (topCandidate) {
+        parentOfTopCandidate = topCandidate.parentNode;
+
+        // Find a better top candidate node if it contains (at least three) nodes which belong to `topCandidates` array
+        // and whose scores are quite closed with current `topCandidate` node.
+        var alternativeCandidateAncestors = [];
+        for (var i = 1; i < topCandidates.length; i++) {
+          if (topCandidates[i].readability.contentScore / topCandidate.readability.contentScore >= 0.75) {
+            alternativeCandidateAncestors.push(this._getNodeAncestors(topCandidates[i]));
+          }
+        }
+        var MINIMUM_TOPCANDIDATES = 3;
+        if (alternativeCandidateAncestors.length >= MINIMUM_TOPCANDIDATES) {
+          while (parentOfTopCandidate.tagName !== "BODY") {
+            if (!parentOfTopCandidate.readability) {
+              parentOfTopCandidate = parentOfTopCandidate.parentNode;
+              continue;
+            }
+            var listsContainingThisAncestor = 0;
+            for (var ancestorIndex = 0; ancestorIndex < alternativeCandidateAncestors.length; ancestorIndex++) {
```

Comment:
How often is the `alternativeCandidateAncestors` array "long"? In my previous review I think I added the `listsContainingThisAncestor >= MINIMUM_TOPCANDIDATES` condition to the loop, which would avoid checking the remaining lists once you've seen the ancestor as relevant for enough candidates, which seems like it'd make sense? Though thinking about it, I guess the condition should be inverted in the loop (ie `listsContainingThisAncestor < MINIMUM_TOPCANDIDATES`)